pkgs/servers/389-directory-server/default.nix

```diff
@@ -0,0 +1,116 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, autoconf
+, automake
+, libtool
+, pkg-config
+, cracklib
+, lmdb
+, json_c
+, linux-pam
+, libevent
+, nspr
+, nss
+, openldap
+, withOpenldap ? true
+, db
+, withBdb ? true
+, cyrus_sasl
+, icu
+, net-snmp
+, withNetSnmp ? true
+, krb5
+, pcre2
+, python3
+, rustPlatform
+, openssl
+, systemd
+, withSystemd ? stdenv.isLinux
+, zlib
+, rsync
+}:
+
+stdenv.mkDerivation rec {
+  pname = "389-directory-server";
+  version = "2.3.0";
+
+  src = fetchFromGitHub {
+    owner = "389ds";
+    repo = "389-ds-base";
+    rev = "389-ds-base-${version}";
+    sha256 = "sha256-GnntF0UaufDrgcM6FFFdwxwVoU9Hu2NXTW1A2lTb6T4=";
+  };
+
+  cargoDeps = rustPlatform.importCargoLock {
+    lockFile = "${src}/src/Cargo.lock";
+  };
+
+  nativeBuildInputs = [
+    autoconf
+    automake
+    libtool
+    pkg-config
+    python3
+    rustPlatform.rust.cargo
+    rustPlatform.rust.rustc
+    rsync
+  ];
+
+  buildInputs = [
+    cracklib
+    lmdb
+    json_c
+    linux-pam
+    libevent
+    nspr
+    nss
+    cyrus_sasl
+    icu
+    krb5
+    pcre2
+    openssl
+    zlib
+  ]
+  ++ lib.optional withSystemd systemd
+  ++ lib.optional withOpenldap openldap
+  ++ lib.optional withBdb db
+  ++ lib.optional withNetSnmp net-snmp;
+
+  postPatch = ''
+    patchShebangs .
+  '';
+
+  preConfigure = ''
+    ./autogen.sh --prefix="$out"
+  '';
+
+  preBuild = ''
+    ln -s ${cargoDeps} /build/source/vendor
+  '';
+
+  configureFlags =
+    [
+      "--enable-rust-offline"
+    ]
+    ++ lib.optionals withSystemd [
+      "--with-systemd"
+      "--with-systemdsystemunitdir=$(out)/etc/systemd/system"
+    ] ++ lib.optionals withOpenldap [
+      "--with-openldap"
+    ] ++ lib.optionals withBdb [
+      "--with-db-inc=${db.dev}/include"
+      "--with-db-lib=${db.out}/lib"
+    ] ++ lib.optionals withNetSnmp [
+      "--with-netsnmp-inc=${net-snmp.dev}/include"
+      "--with-netsnmp-lib=${net-snmp.lib}/lib"
+    ];
+
+  meta = with lib; {
+    description = "An enterprise-class LDAP server for Linux developed by Red Hat as part of the community-supported Fedora Project";
+    homepage = "https://www.port389.org/";
+    license = licenses.asl20;
+    platforms = platforms.linux;
+    maintainers = [ maintainers.ners ];
```

Comment:
Please add yourself as a maintainer by adding an alphabetically sorted entry to `maintainers/maintainer-list.nix` and commit this change into a separate commit with the message ``maintainers: add NAME`` into this PR.

Reply:
This was already done in [the other PR](https://github.com/NixOS/nixpkgs/pull/192649). :slightly_smiling_face: